### PR TITLE
Biocollect 98

### PR DESCRIPTION
**Pull request by sat01a**

### CODE REVIEW

File: grails-app/services/au/org/ala/ecodata/ActivityService.groovy [hunk 1]
```diff
@@ -48,6 +48,24 @@ class ActivityService {
...
 
+    /**
+     * Check user is an owner of an activity.
+     *
+     * @param userId user identifier.
+     * @param activityId activity identifier.
+     * @return
+     */
+    def isUserOwner(userId, activityId) {
+        def list = Activity.createCriteria().list() {
```
Comment (m-r-c) [inline grails-app/services/au/org/ala/ecodata/ActivityService.groovy]: You can change this to 
```
boolean isUserOwner(String userId, String activityId) {
  Activity.countByUserIdAndActivityId(userId, activityId) > 0
}
```

File: grails-app/controllers/au/org/ala/ecodata/ActivityController.groovy [hunk 2]
```diff
@@ -181,6 +181,27 @@ class ActivityController {
...
+    def isUserOwnerForActivity() {
+        String userId = params.userId
+        String activityId = params.activityId
+
+        if (userId && activityId) {
+            boolean isOwner = activityService.isUserOwner(userId, activityId)
+            if (isOwner) {
+                render([userIsOwner: isOwner] as JSON)
+            } else {
+                render status: 404, text: "Activity not found for activityId: ${activityId}"
```
Comment (m-r-c) [inline grails-app/controllers/au/org/ala/ecodata/ActivityController.groovy]: Should it return a 404 if the user is not the owner?

File: grails-app/services/au/org/ala/ecodata/ActivityService.groovy (hunk 1 above)
Comment (sat01a) [inline grails-app/services/au/org/ala/ecodata/ActivityService.groovy]: done

File: grails-app/controllers/au/org/ala/ecodata/ActivityController.groovy (hunk 2 above)
Comment (sat01a) [inline grails-app/controllers/au/org/ala/ecodata/ActivityController.groovy]: done

**Pull request merged**